elide-core/src/main/java/com/yahoo/elide/core/EntityDictionary.java

```diff
@@ -99,7 +72,17 @@ public EntityDictionary() {
      * @return binding class
      */
     public String getBinding(Class<?> entityClass) {
-        return bindEntityToJsonApi.get(lookupEntityClass(entityClass));
+        return entity(entityClass).jsonApi;
+    }
+
+    /**
+     * Get the Database Manager for an entity
+     *
+     * @param entityClass entity name
+     * @return List of relationship names for entity
+     */
+    public DatabaseManager getDatabaseManager(Class<?> entityClass) {
```

Comment:
Another thought: this could still lead to moments of inconsistency (which we may not be able to solve at this level of the problem). Another potential solution is a macro-level three-phase commit where we somehow can "check" that we believe everyone can commit.